### PR TITLE
[PL-47661] edit docs for SAML, add SCIM reference

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
[PL-47661](https://harness.atlassian.net/browse/PL-47661)

This will be effectively published on: https://developer.harness.io/docs/platform/authentication/single-sign-on-saml/#saml-sso-with-azure

Related tickets:
https://harness.atlassian.net/browse/PL-44160
https://harness.atlassian.net/browse/PL-40307

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screenshot. 

[PL-47661]: https://harness.atlassian.net/browse/PL-47661?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/platform/authentication/single-sign-on-saml.md [hunk 1]
```diff
@@ -20,6 +20,12 @@ The XML SAML file used with Harness must use UTF-8.
...
+
+To ensure continuous and real-time synchronization of user group bindings and access controls, it is recommended to utilize the System for Cross-domain Identity Management (SCIM) protocol. SCIM enables real-time syncing of user additions with Harness user groups, ensuring that user permissions and access rights are consistently applied and maintained. For implementation details and further guidance on provisioning users with SCIM, please refer to the [Okta SCIM](https://developer.harness.io/docs/platform/role-based-access-control/provision-users-with-okta-scim), [Microsoft Entra SCIM](https://developer.harness.io/docs/platform/role-based-access-control/provision-users-and-groups-using-azure-ad-scim), [OneLogin SCIM](https://developer.harness.io/docs/platform/role-based-access-control/provision-users-and-groups-with-one-login-scim).
```
Comment (brian-f-harness) [inline docs/platform/authentication/single-sign-on-saml.md:26]:
```suggestion
To ensure continuous and real-time synchronization of user group bindings and access controls, it is recommended to utilize the System for Cross-domain Identity Management (SCIM) protocol. SCIM enables real-time syncing of user additions with Harness user groups, ensuring that user permissions and access rights are consistently applied and maintained. For implementation details and further guidance on provisioning users with SCIM, go to [Okta SCIM](/docs/platform/role-based-access-control/provision-users-with-okta-scim), [Microsoft Entra SCIM](/docs/platform/role-based-access-control/provision-users-and-groups-using-azure-ad-scim), and [OneLogin SCIM](/docs/platform/role-based-access-control/provision-users-and-groups-with-one-login-scim).
```

**Review by brian-f-harness — APPROVED** [review]
LGTM

**Review by pratmit — APPROVED** [review]
LGTM

**Review by brian-f-harness — APPROVED** [review]

**Comment by saifeemustafaqharness** [comment]
@brian-f-harness Can I please have your review and approval, Thanks

**Review by brian-f-harness — APPROVED** [review]

**Pull request merged**